app/controllers/kernel_builds_controller.rb

```diff
@@ -13,10 +14,14 @@ def show
   # GET /kernel_builds/new
   def new
     @kernel_build = KernelBuild.new
+    @current_user_kernel_configs ||= KernelConfig.where(user_id: current_user.id).distinct
+    @current_user_kernel_sources ||= KernelSource.where(user_id: current_user.id).distinct
```

Comment:
Since `User has_many :kernel_configs`, you can access through the `current_user` object to reduce complexity.
```suggestion
    @current_user_kernel_configs ||= current_user.kernel_configs.distinct
    @current_user_kernel_sources ||= current_user.kernel_sources.distinct
```